databases/Assignment2_answers.sql

```diff
@@ -0,0 +1,316 @@
+FROM Product
+INNER JOIN Computer USING Model
+WHERE Product.Type = laptop AND Computer.RAM > 256 AND Product.Manufacturer = D;
+
+9. Which computer models that have more than 256 MB of RAM
+are produced by manufacturer D?
+SELECT Computer.Model
+FROM Product
+INNER JOIN Computer USING Model
+WHERE Computer.RAM > 256 AND Product.Manufacturer = D;
+
+10. Which laptop models cost less than $1200?
+SELECT Computer.Model
+FROM Product
+INNER JOIN Computer USING Model
+WHERE Product.Type = laptop AND Computer.Price < 1200;
+
+11. Which manufacturers produce personal computers and printers?
+SELECT DISTINCT Manufacturer
+FROM Product
+WHERE Type = pc OR Type = printer;
+
+12. Which manufacturers produce all the product types?
+SELECT Manufacturer
+FROM Product
+GROUP BY Manufacturer
+HAVING COUNT(DISTINCT Type) = 3;
+
+13. Which manufacturers supply only personal computers?
+
+SELECT distinct manufacturers
+FROM Product
+WHERE Type ='pc'
+
+
+
+with pcs AS (
+    SELECT Manufacturer, Model
+    FROM Product
+    WHERE Type = 'pc'
+),
+
+
+
+
+with not_pcs AS (
+    SELECT Manufacturer
+    FROM Product
+    WHERE Type != 'pc'
+    GROUP BY Manufacturer
+)
+SELECT Product.Manufacturer
+FROM Product
+WHERE Manufacturer NOT IN not_pcs;
+
+ALSO:
+
+select all of the ones that ONLY have one type
+intersect
+SELECT all of the ones that are only type= PC
+
+
+# sub query with Type != pc, outer left join it together ...
+
+EXCEPT; IS NOT IN
+
+HAVING SUM(Type) = 'pc';
+
+HAVING Type = 'pc';
+# HAVING MAX(TYPE) = 'pc';
+# the case where manufacturers sell PCs, but other things as well!
+
+
+# HAVING Type = pc .... i dont think this is right, but why?
+
+
+14. Which manufacturers supply only one type of product?
+SELECT Manufactuer
+FROM Product
+GROUP BY Manufactuer
+HAVING COUNT(DISTINCT Type) = 1;
+
+
+
+15. You want to buy: #think about how to intersect, instead of JOINS? which is more efficient? why?
+• a personal computer with at least 1024 MB of RAM and a CD with 56x, and
+• a laptop with a screen larger than 1500, and
+• a printer cheaper than $300
+all produced by the same manufacturer. Which manufacturers should you consider?
+
+with computers AS (
+    SELECT distinct Manufacturer
+    FROM Computer
+    INNER JOIN Product USING Model
+    WHERE Product.Type = pc AND RAM > 1024 AND CD = 56x;
+),
+laptops AS (
+    SELECT distinct Manufactuer
+    FROM Computer
+    INNER JOIN Product USING Model
+    WHERE Product.Type = laptop AND Computer.Screen > 15;
+),
+printers AS (
+    SELECT distinct Manufacturer
+    FROM Printer
+    INNER JOIN Product USING Model
+    WHERE Price < 300;
+)
+SELECT c.Manufacturer
+FROM computers c
+INNER JOIN laptops l USING Manufacturer
+INNER JOIN printers p USING Manufacturer
+;
+
+
+
+
+# do sequential JOINS happen simultaneously?
+SELECT DISTINCT Product.Manufacturer
+FROM Product
+INNER JOIN Computer USING Model
+WHERE Product.Type = pc AND Computer.RAM > 1024 AND Computer.CD = 56x
+INNER JOIN Computer USING Model
```

Comment:
To be clear, you can do multiple joins in a query, but only one WHERE.

Also, I suppose this could technical work if you moved the WHERE conditions into the ON condition of each JOIN, but I think that's poor form. Though it would still give the right answer.